Python/optimizer_bytecodes.c

```diff
@@ -1238,6 +1230,98 @@ dummy_func(void) {
         }
     }
 
+    op(_GUARD_GLOBALS_VERSION, (version/1 --)) {
+        if (ctx->frame->func != NULL) {
+            PyObject *globals = ctx->frame->func->func_globals;
+            if (incorrect_keys(globals, version)) {
+                OPT_STAT_INC(remove_globals_incorrect_keys);
+                ctx->done = true;
+            }
+            else if (get_mutations(globals) >= _Py_MAX_ALLOWED_GLOBALS_MODIFICATIONS) {
+                /* Do nothing */
+            }
+            else {
+                if (!ctx->frame->globals_watched) {
+                    PyDict_Watch(GLOBALS_WATCHER_ID, globals);
+                    _Py_BloomFilter_Add(dependencies, globals);
+                    ctx->frame->globals_watched = true;
+                }
+                if (ctx->frame->globals_checked_version == version) {
+                    REPLACE_OP(this_instr, _NOP, 0, 0);
+                }
+            }
+        }
+        ctx->frame->globals_checked_version = version;
+    }
+
+    op(_LOAD_GLOBAL_BUILTINS, (version/1, index/1 -- res))
+    {
+        (void)version;
+        (void)index;
+        PyObject *cnst = NULL;
+        PyInterpreterState *interp = _PyInterpreterState_GET();
+        PyObject *builtins = interp->builtins;
+        if (incorrect_keys(builtins, version)) {
+            OPT_STAT_INC(remove_globals_incorrect_keys);
+            ctx->done = true;
+        }
+        else if (interp->rare_events.builtin_dict >= _Py_MAX_ALLOWED_BUILTINS_MODIFICATIONS) {
+            /* Do nothing */
+        }
+        else {
+            if (!ctx->frame->builtins_watched) {
+                PyDict_Watch(BUILTINS_WATCHER_ID, builtins);
+                ctx->frame->builtins_watched = true;
+            }
+            if (ctx->frame->globals_checked_version != 0 && ctx->frame->globals_watched) {
+                cnst = convert_global_to_const(this_instr, builtins, false);
+            }
+        }
+        if (cnst == NULL) {
+            res = sym_new_unknown(ctx);
```

Comment:
Isn't this guaranteed to not be NULL at least?
```suggestion
            res = sym_new_non_null(ctx);
```